lib/config/presets/index.ts

```diff
@@ -218,15 +219,24 @@ export async function getPreset(
   }
   const { presetSource, repo, presetPath, presetName, tag, params } =
     parsePreset(preset);
-  let presetConfig = await presetSources[presetSource].getPreset({
-    repo,
-    presetPath,
-    presetName,
-    tag,
-  });
+
+  const cacheKey = preset;
+  const cachedResult = memCache.get<Preset>(cacheKey);
+  let presetConfig = is.undefined(cachedResult)
+    ? await presetSources[presetSource].getPreset({
+        repo,
+        presetPath,
+        presetName,
+        tag,
+      })
+    : cachedResult;
```

Comment:
```suggestion
  const cacheKey = preset;
  const cachedResult = memCache.get<Preset>(cacheKey);
  if (cachedResult) {
    return cachedResult;
  }

  let presetConfig = await presetSources[presetSource].getPreset({
        repo,
        presetPath,
        presetName,
        tag,
      });
```
simpler logic, probably needs prettify